src/apps/concepts/components/ConceptsTableRow.tsx

```diff
@@ -0,0 +1,296 @@
+import React from "react";
+import { APIConcept } from "../types";
+import {
+  TableRow,
+  TableCell,
+  Checkbox,
+  Tooltip,
+  IconButton,
+} from "@material-ui/core";
+import { MoreVert as MoreVertIcon } from "@material-ui/icons";
+import { Link } from "react-router-dom";
+import { ConceptsActionMenu } from "./ConceptsActionMenu";
+
+export function showEditMenuItem(
+  concept: APIConcept,
+  showingEditButtons: boolean,
+  linkedSource: string | undefined,
+  canModifyConcept: (concept: APIConcept) => boolean
+) {
+  // only allow edit of a concept we can modify and belongs to our linked source
+  // the second condition prevents us editing non custom concepts in a collection
+  return (
+    showingEditButtons &&
+    canModifyConcept(concept) &&
+    linkedSource &&
+    concept.url.includes(linkedSource)
+  );
+}
+
+export function showRemoveFromDictionaryMenuItem(
+  concept: APIConcept,
+  showingEditButtons: boolean,
+  linkedSource: string | undefined
+) {
+  // only allow manual removal of imported/ non custom concepts
+  return (
+    showingEditButtons && linkedSource && !concept.url.includes(linkedSource)
+  );
+}
+/*
+  Show the action icon only when the concept has 
+  atleast one of the 3 options - edit, remove or add to Dictionary
+  */
+export function showActionIcon(
+  concept: APIConcept,
+  buttons: { [key: string]: boolean },
+  linkedSource: string | undefined,
+  canModifyConcept: (concept: APIConcept) => boolean
+) {
+  return (
+    showEditMenuItem(concept, buttons.edit, linkedSource, canModifyConcept) ||
+    showRemoveFromDictionaryMenuItem(concept, buttons.edit, linkedSource) ||
+    buttons.addToDictionary
+  );
+}
+
+const actionIcon = (
+  index: number,
+  toggleMenu: (
+    index: number,
+    event?: React.MouseEvent<HTMLButtonElement, MouseEvent> | undefined
+  ) => void
+) => {
+  return (
+    <Tooltip title='More actions' enterDelay={700}>
+      <IconButton
+        data-testid={"action-icon"}
+        id={`${index}.menu-icon`}
+        aria-controls={`${index}.menu`}
+        aria-haspopup='true'
+        onClick={(event) => toggleMenu(index, event)}
+      >
+        <MoreVertIcon />
+      </IconButton>
+    </Tooltip>
+  );
+};
+
+const conceptNameCell = (
+  toggleSelect: (event: React.MouseEvent<unknown>, id: string) => void,
+  row: APIConcept,
+  classes: Record<
```

Comment:
@kirity-mahanty didn't understand the classes part. We can have a quick catchup on this

Reply:
Good catch. During refactoring, the tool defined the type of `classes` parameter like this. It is used for styling. I thought to modify those but missed.
I removed that parameter from some of the functional components as it is redundant in many calls. Rather I added the relevant styles in `index.scss` file. We can discuss this if required.